File: app/src/main/java/org/kamwas/android_test/MainActivity.java
package org.kamwas.android_test;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class MainActivity extends AppCompatActivity {

    private TextView textView2;
    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView2 = findViewById(R.id.textView2);
        button = findViewById(R.id.button);

        CollectionPerformance collectionPerformance = new CollectionPerformance();
        button.setOnClickListener(b -> {
            textView2.setText(collectionPerformance.sort() + "");
        });

    }
}

File: kotlin/app/src/main/java/kotlin_test/MainActivity.kt
package kotlin_test

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        buttonMenu(collectionButton, CollectionActivity::class.java)
        buttonMenu(restButton, RESTActivity::class.java)
        buttonMenu(dbButton, DBActivity::class.java)
        buttonMenu(deserButton, DeSerActivity::class.java)
        buttonMenu(fileButton, FileActivity::class.java)
    }

    private fun buttonMenu(button: Button, clz: Class<out AppCompatActivity>) {
        button.setOnClickListener { startActivity(Intent(applicationContext, clz)) }
    }

}